Edit an AWS Account

thibautcornolti opened the pull request:
Added PATCH method to the /aws route to edit an AWS Account. Only the pretty field can be edited.

schubev commented on api/files/api/app/models.py:
```diff
@@ -178,10 +179,16 @@ def get_boto_session(self):
+                self.is_valid_key = True
+                db.session.commit()
+        except botocore.exceptions.ClientError, e:
+            if e.response['Error']['Code'] == 'InvalidClientTokenId':
```
Catching the expression this way means other `botocore.exceptions.ClientError` which do not have error code `InvalidClientTokenId` will be silenced. You can do that with the `raise` statement (with no argument).

thibautcornolti committed:
aws: add the PATCH method to /aws

The PATCH method was added in the files aws/routeAwsPatch.go
and aws/routes.go to edit an AWS Account.

TRAC-575 #comment PATCH method added to edit an AWS Account

giubil commented on aws/aws.go:
```diff
@@ -137,6 +137,20 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdateAWSAccount(ctx context.Context, tx *sql.Tx) error {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	dbAwsAccount, _ := models.AwsAccountByID(tx, a.Id)
```
Why don't you check the error return value of `models.AwsAccountByID()` ?

giubil commented on aws/routeAwsPatch.go:
```diff
@@ -0,0 +1,70 @@
+	if err == nil {
+		awsAccount.Pretty = body.Pretty
+		if err := awsAccount.UpdateAWSAccount(ctx, tx); err != nil {
+			return 500, errFailUpdateAccount
```
You should log an error here

thibautcornolti committed:
aws: add logger to the way to update an AWS Account

schubev commented on aws/aws.go:
```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
+// UpdateAWSAccount updates an AWS account for a user. It does no error
+// checking: the caller should check themselves that the AWS account exists.
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdateAWSAccount(ctx context.Context, tx *sql.Tx) error {
```
If only the `pretty` is updated, the function’s name should reflect it.

schubev commented on aws/aws.go:
```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
+		logger.Error("Failed to get AWS account in database.", nil)
+	} else {
+		dbAwsAccount.Pretty = a.Pretty
+		err := dbAwsAccount.Update(tx)
```
By using `:=` here you shadow the `err` defined at line 145, so at line 155 you would still return `nil`.

schubev replied:
Please write unit tests, you should have caught this with unit tests.

thibautcornolti replied:
`err` is shadowed only if the `err` defined at line 145 is `nil`. It's the expected behaviour.

schubev commented on aws/routes.go:
```diff
@@ -27,6 +27,13 @@ import (
+	// QueryArgAwsAccount allows to get the AWS Account ID in the URL Parameters
+	// with routes.RequiredQueryArgs. This AWS Account ID will be an Uint stored in
+	// the routes.Arguments map with itself for key.
+	QueryArgAwsAccount = routes.QueryArg{"aa", "AWS Account ID", routes.QueryArgUint{}}
```
Isn’t that already defined somewhere else ?

schubev reviewed (dismissed):
Beware of variable shadowing.

thibautcornolti committed:
aws: Minor change in the PATCH method

Function renamed to be more explicit
Add errors in log
Remove duplicated variable

TRAC-572

schubev commented on aws/aws.go:
```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
+// UpdatePrettyAWSAccount updates an AWS account for a user. It does no error
+// checking: the caller should check themselves that the AWS account exists.
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdatePrettyAWSAccount(ctx context.Context, tx *sql.Tx) error {
```
The convention in this project is to treat acronyms as simple words, with only the first letter capitalized, so this should be `UpdatePrettyAwsAccount`.

schubev reviewed (changes requested):
Just one change to an identifier, and you’re good !

thibautcornolti committed:
aws: Change function name

TRAC-572
